dev/tools/merge-pr.sh

```diff
@@ -107,6 +110,25 @@ if [ "$BASE_BRANCH" != "coq:$CURRENT_LOCAL_BRANCH" ]; then
   ask_confirmation
 fi;
 
+# Sanity check: the local branch is up-to-date with upstream
+
+LOCAL_BRANCH_COMMIT=$(git rev-parse HEAD)
+UPSTREAM_COMMIT=$(git rev-parse @{u})
+if [ "$LOCAL_BRANCH_COMMIT" != "$UPSTREAM_COMMIT" ]; then
+
+    # Is it just that the upstream branch is behind?
+    # It could just be that we merged other PRs and we didn't push yet
+
+    if git merge-base --is-ancestor -- "$UPSTREAM_COMMIT" "$LOCAL_BRANCH_COMMIT"; then
+        warning "Your branch is ahead of ${REMOTE}."
+        warning "You should see this warning only if you just merged another PR and did not push yet."
+    else
+        error "Local branch is not up-to-date with ${REMOTE}."
+        error "Pull before merging."
+        ask_confirmation
```

Comment:
Here we should not ask for confirmation, we should unconditionally error. If we merged and pushed we could lose commits.

Reply:
Well, you would need to force-push to do that. And force-pushing is actually disabled for everyone (including admins) on our branches. However, we can imagine a use-case (if someone uses this script to merge in a branch in their fork and they want to force-push...).